template.rb

```diff
@@ -40,31 +32,42 @@ def apply_template!
   git :init unless preexisting_git_repo?
   empty_directory ".git/safe"
 
-  run_with_clean_bundler_env "bundle update"
-  run_with_clean_bundler_env "bin/rails webpacker:install"
-  install_dart_sass unless sprockets?
-  create_database_and_initial_migration
-  run_with_clean_bundler_env "bin/setup"
-
-  binstubs = %w[brakeman bundler bundler-audit rubocop sidekiq]
-  run_with_clean_bundler_env "bundle binstubs #{binstubs.join(' ')} --force"
-
-  template "rubocop.yml.tt", ".rubocop.yml"
-  run_rubocop_autocorrections
-
-  template "eslintrc.js", ".eslintrc.js"
-  template "prettierrc.js", ".prettierrc.js"
-  template "stylelintrc.js", ".stylelintrc.js"
-  add_yarn_lint_and_run_fix
-  add_yarn_start_script
-
-  unless any_local_git_commits?
-    git checkout: "-b main"
-    git add: "-A ."
-    git commit: "-n -m 'Set up project'"
-    if git_repo_specified?
-      git remote: "add origin #{git_repo_url.shellescape}"
-      git push: "-u origin --all"
+  after_bundle do
+    apply "app/assets/template.rb"
+    apply "app/template.rb"
+
+    run_with_clean_bundler_env "bundle update"
+    create_database_and_initial_migration
+    run_with_clean_bundler_env "bin/setup"
+
+    binstubs = %w[brakeman bundler bundler-audit rubocop sidekiq]
+    run_with_clean_bundler_env "bundle binstubs #{binstubs.join(' ')} --force"
+
+    if File.exist?("bin/dev")
+      append_to_file "Procfile.dev", "worker: bin/sidekiq"
+    else
+      remove_file "Procfile.dev"
+    end
+
+    template "rubocop.yml.tt", ".rubocop.yml"
+    run_rubocop_autocorrections
+
+    template "eslintrc.js", ".eslintrc.js"
+    template "prettierrc.js", ".prettierrc.js"
+    template "stylelintrc.js", ".stylelintrc.js"
+    add_yarn_lint_and_run_fix
+    add_yarn_start_script
+
+    append_to_file ".gitignore", "node_modules" unless File.read(".gitignore").match?(%{^/?node_modules})
```

Comment:
By default, Rails 7 does not use any Node packages unless you explicitly opt in. That means `node_modules` won't be in the default `.gitignore`. But we add our own Node packages for things like eslint. So we need to make sure `node_modules` is properly ignored in that case.